include/zephyr/drivers/spi.h

```diff
@@ -466,7 +475,8 @@ typedef int (*spi_api_io_async)(const struct device *dev,
 				const struct spi_config *config,
 				const struct spi_buf_set *tx_bufs,
 				const struct spi_buf_set *rx_bufs,
-				struct k_poll_signal *async);
+				spi_callback_t cb,
```

Comment:
There is no gain in merging both cb and data pointer into a dedicated struct?

Reply:
From an API perspective this feels pretty natural and follows other callback like APIs in Zephyr, not sure there's a strong reason to package it up in a struct first

Reply:
ok